### PR TITLE
Fix/remove renew

### PR DESCRIPTION
**Pull request by rkallos**
This PR removes renew capability from canal. In my opinion, it was not implemented well (for the simple reason that the callbacks could not know whether a call to renew or reauth failed or not), and it was buggy. We don't need it right now, and should we ever need it in the future, we should try and implement it better.

### CODE REVIEW

File: src/canal_utils.erl [hunk 1]
```diff
@@ -15,6 +18,13 @@ error_msg(Msg) ->
     error_logger:error_report(Msg).
 
 
+-spec error_msg(list(), list()) -> ok.
```
Comment (gnuvince) [inline src/canal_utils.erl]: `io_lib:format`'s first argument is of type `io:format()` (an alias for `atom() | string() | binary()`).  Does it make sense to change your spec to be in line with the function you call?

File: include/canal_internal.hrl [hunk 2]
```diff
@@ -13,8 +13,5 @@
 % types
 
 -type req_id()  :: shackle:request_id().
--type req() :: {{pid(), reference()}, write | read} | renew.
+-type req() :: {{pid(), reference()}, write | read}.
```
Comment (gnuvince) [inline include/canal_internal.hrl:16]: Do you need to add `reauth` to `req()`?

**Review by gnuvince — APPROVED** [review]
From a cursory examination, this looks okay.  I left a couple of minor comments.  Good luck!

File: include/canal_internal.hrl (hunk 2 above)
Comment (rkallos) [reply include/canal_internal.hrl:16]: Thanks for checking! Authentication is done synchronously within the server code, unlike read and write requests, so it doesn't need to be added here.

File: src/canal_utils.erl (hunk 1 above)
Comment (rkallos) [reply src/canal_utils.erl]: Fixed. Thank you

File: src/canal.erl [hunk 3]
```diff
@@ -182,11 +146,9 @@ handle_info({#cast{request_id = RequestId}, Error = {error, _}}, State) ->
         {read, From} ->
             gen_server:reply(From, Error);
         {write, From} ->
-            gen_server:reply(From, Error);
-        {renew, _} ->
-            ok
+            gen_server:reply(From, Error)
     end,
-
+    % blow up here?
```
Comment (RAttab) [inline src/canal.erl]: There's a lot of these added in the PR. Why the uncertainty?

**Comment by rkallos** [comment]
Shucks, those comments shouldn't be there. I'll fix that shortly.

**Review by FrankBro — APPROVED** [review]

**Pull request merged**